nodes/usb_cam_node.cpp

```diff
@@ -326,10 +326,16 @@ class UsbCamNode
     ROS_ASSERT(diag_freq_camera_info_);
 
     enable_auto_reset_exposure_ = true;
-    auto_reset_exposure_timer_ = node_.createTimer(
-      ros::Duration(auto_reset_exposure_period_),
-      boost::bind(&UsbCamNode::checkAutoResetExposure, this, _1)
-    );
+    if (auto_reset_exposure_period_ > 0) {
+      auto_reset_exposure_timer_ = node_.createTimer(
+          ros::Duration(auto_reset_exposure_period_),
+          boost::bind(&UsbCamNode::checkAutoResetExposure, this, _1));
+    } else {
+      // Reset exposure once just in case it wasn't applied
+      std::this_thread::sleep_for(
+          std::chrono::seconds{WAIT_CHANGING_AUTO_EXPOSURE_SEC});
+      resetExposureSettings();
+    }
```

Comment:
I'd say let's make a oneshot timer for this instead of sleeping in the thread

Reply:
Oh nice I didn't know that is an arg for the timer. Fixed.